tasks/utils/artifact.js

```diff
@@ -98,18 +92,21 @@ class RequestStore {
     await this.writeFile(id, mergedData)
   }
 
-  async upsert(id /*: string*/, data /*: Fragment<RequestData>*/) /*: Promise<void> */ {
+  async upsert(id /*: string*/, data /*: Fragment<RequestData>*/) /*: Promise<boolean> */ {
+    let created = false
     let previousData = {}
     let newData = data
     try {
       previousData = await this.read(id)
     } catch {
+      created = true
```

Comment:
Do we always expect an exception here?

Reply:
Yes, `.read` will throw if the file is not found.